### PR TITLE
FWI-3435: Docs for validate rules

### PR DESCRIPTION
**Pull request by jdesouza**
This PR fixes #

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### CODE REVIEW

File: docs/configure/cli/automation-rules.md [hunk 1]
```diff
@@ -88,3 +88,45 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
...
+
+```bash
+insights-cli validate rule -t  <insights context> {-r <rule file> -a <action item file>} [-i <expected output file>]
+```
+
+Parameters:
+
+Insights context: possible input: AdmissionController, Agent, CI/CD
+Report type: possible input: opa, nova, kubesec, kube-hunter, kube-bench, goldilocks, admission, pluto, polaris, rbac-reporter, release-watcher, prometheus-metrics, resource-metrics, trivy, workloads, right-sizer, awscosts, falco
+You can also provide different file input path, rule path and the expected output.
```
Comment (rbren) [inline docs/configure/cli/automation-rules.md:127]:
```suggestion
```
I'm a little confused by this part--I think maybe we should remove, so we don't have to keep it up to date. Maybe include it in an example?

File: docs/configure/cli/automation-rules.md [hunk 2]
```diff
@@ -88,3 +88,45 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
...
+```yaml
+title: Image has vulnerabilities
+cluster: production
+severity: 0.9
+```
+
+Once the files have been created, use the following command to validate the Rules to Insights
+
+```bash
+insights-cli validate rule -t  <insights context> {-r <rule file> -a <action item file>} [-i <expected output file>]
```
Comment (rbren) [inline docs/configure/cli/automation-rules.md:120]: can you use the full flag names, so it's obvious what they do? Ditto for the example.

File: docs/configure/cli/automation-rules.md (hunk 1 above)
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md]: done

File: docs/configure/cli/automation-rules.md (hunk 2 above)
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md]: done

File: docs/configure/cli/automation-rules.md [hunk 3]
```diff
@@ -88,3 +88,50 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
 For additional information see
 * [OPA policies With CLI](/configure/cli/opa)
 * [Policies Configuration with CLI](/configure/cli/settings)
+
+## Testing Automation Rules before inserting into Insights
+Before pushing Automation Rules to Insights, you can use the CLI to check if the results will be as expected:
```
Comment (lizcoolman) [inline docs/configure/cli/automation-rules.md:93]: Before pushing Automation Rules to Insights, you can use the CLI to check if the results will work as expected:

File: docs/configure/cli/automation-rules.md [hunk 4]
```diff
@@ -88,3 +88,50 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
...
+
+```bash
+insights-cli validate rule --insights-context  <insights context> --report-type <report type> {--automation-rule-file <rule file> --action-item-file <action item file>} [--expected-action-item <expected output file>]
+```
+Example:
+```bash
+insights-cli validate rule --insights-context Agent --report-type trivy --automation-rule-file ./rule.js --action-item-file ./action-items.yaml --expected-action-item ./expected-output.yaml
+```
+
+if expected output is provided and the result is the expected one, a success message is displayed:
```
Comment (lizcoolman) [inline docs/configure/cli/automation-rules.md:127]: Update to: "If the expected output is provided and the actual result matches, a success message is displayed:"

**Review by lizcoolman — CHANGES_REQUESTED** [review]
Added comments with proposed changes

File: docs/configure/cli/automation-rules.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md]: done

File: docs/configure/cli/automation-rules.md [hunk 5]
```diff
@@ -88,3 +88,50 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
 For additional information see
 * [OPA policies With CLI](/configure/cli/opa)
 * [Policies Configuration with CLI](/configure/cli/settings)
+
+## Testing Automation Rules before inserting into Insights
```
[the remark this answers is not in the document]
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md:92]: done

File: docs/configure/cli/automation-rules.md [hunk 6]
```diff
@@ -88,3 +88,50 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
...
+```
+
+Expected output Action item file example:
+```yaml
+title: Image has vulnerabilities
+cluster: production
+severity: 0.9
+```
+
+Once the files have been created, use the following command to validate the rule against Insights
```
[the remark this answers is not in the document]
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md:117]: done

File: docs/configure/cli/automation-rules.md (hunk 4 above)
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md]: done

File: docs/configure/cli/automation-rules.md [hunk 7]
```diff
@@ -88,3 +88,50 @@ The `--delete` flag is not available for the `push all` command to avoid unexpec
 For additional information see
 * [OPA policies With CLI](/configure/cli/opa)
 * [Policies Configuration with CLI](/configure/cli/settings)
+
+## Testing Automation Rules before inserting into Insights
+Before pushing Automation Rules to Insights, you can use the CLI to check if the results will be as expected:
+For the testing we need a rule file and action item file. Optionally we can provide an expected output file to compare:
```
[the remark this answers is not in the document]
Comment (jdesouza) [reply docs/configure/cli/automation-rules.md]: Right. Done

**Review by lizcoolman — APPROVED** [review]

**Pull request merged**